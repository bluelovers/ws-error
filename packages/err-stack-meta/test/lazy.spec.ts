import errStackMeta from '../src/index';
import { basename } from 'path';

describe(basename(__filename), () =>
{

	test(`test`, () =>
	{
		let e = new AggregateError([new Error('1'), new Error('3')], 'ggregateError:\n' +
			'    at Object');

		_check(e);

	});

	test(`test2`, () =>
	{
		let e = new AggregateError([new Error('1'), new Error('3')], 'ggregateError:');

		_check(e);

	});

	test(`test3`, () =>
	{
		let e = new AggregateError([new Error('1'), new Error('3')], 'AggregateError:');

		_check(e);

	});

	test(`test4`, () =>
	{
		let e = new AggregateError([new AggregateError([new Error('1'), new Error('3')]), new Error('3')], 'AggregateError:');

		e.name = 'AggregateError2';

		_check(e);

	});

})

function _check(e: Error)
{
	let actual = errStackMeta(e);

	expect(actual.prefix).toContain(e.name);
	expect(actual.stack.length).toBeGreaterThan(0);
	expect(actual.stack).toMatch(/^\s+at /);

	expect(actual).toMatchSnapshot({
		stack: expect.any(String),
	});
}
