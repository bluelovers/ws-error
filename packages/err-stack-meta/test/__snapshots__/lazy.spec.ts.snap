// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy.spec.ts test 1`] = `
{
  "error": [AggregateError: ggregateError:
    at Object],
  "message": "ggregateError:
    at Object",
  "prefix": "AggregateError: ",
  "rawTrace": Any<Array>,
  "stack": Any<String>,
  "type": "AggregateError",
}
`;

exports[`lazy.spec.ts test2 1`] = `
{
  "error": [AggregateError: ggregateError:],
  "message": "ggregateError:",
  "prefix": "AggregateError: ",
  "rawTrace": Any<Array>,
  "stack": Any<String>,
  "type": "AggregateError",
}
`;

exports[`lazy.spec.ts test3 1`] = `
{
  "error": [AggregateError: AggregateError:],
  "message": "AggregateError:",
  "prefix": "AggregateError: ",
  "rawTrace": Any<Array>,
  "stack": Any<String>,
  "type": "AggregateError",
}
`;

exports[`lazy.spec.ts test4 1`] = `
{
  "error": [AggregateError2: AggregateError:],
  "message": "AggregateError:",
  "prefix": "AggregateError2: ",
  "rawTrace": Any<Array>,
  "stack": Any<String>,
  "type": "AggregateError2",
}
`;
